src/main/cli/cli.c

```diff
@@ -6121,120 +6220,169 @@ static void cliHelp(char *cmdline)
             return;
-        } else if (c == 12) {                  // NewPage / CTRL-L
-            // clear screen
-            cliPrint("\033[2J\033[1;1H");
-            cliPrompt();
-        } else if (bufferIndex && (c == '\n' || c == '\r')) {
-            // enter pressed
-            cliPrintLinefeed();
+        }
 
-            // Strip comment starting with # from line
-            char *p = cliBuffer;
-            p = strchr(p, '#');
-            if (NULL != p) {
-                bufferIndex = (uint32_t)(p - cliBuffer);
-            }
+        cliPrompt();
+    } else if (c == 127) {
+        // backspace
+        if (bufferIndex) {
+            cliBuffer[--bufferIndex] = 0;
+            cliPrint("\010 \010");
+        }
+    } else if (bufferIndex < sizeof(cliBuffer) && c >= 32 && c <= 126) {
+        if (!bufferIndex && c == ' ')
+            return; // Ignore leading spaces
+        cliBuffer[bufferIndex++] = c;
+        cliWrite(c);
+    }
+}
 
-            // Strip trailing whitespace
-            while (bufferIndex > 0 && cliBuffer[bufferIndex - 1] == ' ') {
-                bufferIndex--;
-            }
+void cliProcess(void)
+{
+    if (!cliWriter) {
+        return;
+    }
 
-            // Process non-empty lines
-            if (bufferIndex > 0) {
-                cliBuffer[bufferIndex] = 0; // null terminate
+    // Be a little bit tricky.  Flush the last inputs buffer, if any.
+    cliWriterFlush();
 
-                const clicmd_t *cmd;
-                char *options;
-                for (cmd = cmdTable; cmd < cmdTable + ARRAYLEN(cmdTable); cmd++) {
-                    if ((options = checkCommand(cliBuffer, cmd->name))) {
-                        break;
-                    }
-                }
-                if (cmd < cmdTable + ARRAYLEN(cmdTable)) {
-                    cmd->func(options);
-                } else {
-                    cliPrintError("UNKNOWN COMMAND, TRY 'HELP'");
-                }
-                bufferIndex = 0;
-            }
+    while (serialRxBytesWaiting(cliPort)) {
+        uint8_t c = serialRead(cliPort);
 
-            memset(cliBuffer, 0, sizeof(cliBuffer));
+        processCharacter(c);
+    }
+}
 
-            // 'exit' will reset this flag, so we don't need to print prompt again
-            if (!cliMode)
-                return;
+#if defined(USE_CUSTOM_DEFAULTS)
+void cliProcessDefaults(void)
+{
+    if (processingDefaults) {
+        return;
+    }
 
-            cliPrompt();
-        } else if (c == 127) {
-            // backspace
-            if (bufferIndex) {
-                cliBuffer[--bufferIndex] = 0;
-                cliPrint("\010 \010");
-            }
-        } else if (bufferIndex < sizeof(cliBuffer) && c >= 32 && c <= 126) {
-            if (!bufferIndex && c == ' ')
-                continue; // Ignore leading spaces
-            cliBuffer[bufferIndex++] = c;
-            cliWrite(c);
+    char *ptr = defaultsStart;
+    if (isDefaults(ptr)) {
```

Comment:
can the method name be improved here? `isDefaults()` is kinda vague.  maybe `isCustomDefaultsBlock()` ?

Reply:
actaully now I think of of it, prefixing 'defaults' that's used a lot in this PR with the word 'custom', i.e. 'customDefaults' works a lot better.

e.g.  
```
cliProcessCustomDefaults()...
if (processingCustomDefaults)...
__customDefaults_start
```
etc.

Reply:
Good point, will change.

Reply:
the naming `isDefaults` still needs doing.  `isCustomDefaults`.